DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,557 @@
+	clothing_flags = SHOWEROKAY | MASKINTERNALS
+	var/cooldown2 = 60 //6 second cooldown
+	var/saved_time = 0
+	actions_types = list(/datum/action/item_action/futile, /datum/action/item_action/message_collective)
+
+/datum/action/item_action/message_collective
+	name = "Message the collective"
+	icon_icon = 'DS13/icons/actions/actions_borg.dmi'
+	button_icon_state = "message_collective"
+
+/datum/action/item_action/message_collective/Trigger()
+	var/obj/item/clothing/mask/gas/borg/FT = target
+	if(istype(FT))
+		FT.message_collective()
+	return ..()
+
+/obj/item/clothing/mask/gas/borg/AltClick(mob/user)
+	futile()
+
+/obj/item/clothing/mask/gas/borg/CtrlClick(mob/user)
+	futile()
+
+/obj/item/clothing/mask/gas/borg/ui_action_click(mob/user, actiontype)
+	if(actiontype == /datum/action/item_action/futile)
+		futile(user)
+	else if(actiontype == /datum/action/item_action/message_collective)
+		message_collective(user)
+
+/obj/item/clothing/mask/gas/borg/proc/message_collective(var/mob/living/carbon/human/user)
+	if(!user)
+		user = src.loc
+		if(!istype(user, /mob))
+			return
+	if(user.stat == DEAD || user.IsUnconscious())
+		return
+	var/message = stripped_input(user,"Communicate with the collective.","Send Message")
+	if(!message)
+		return
+	GLOB.borg_collective.message_collective(user.real_name, message)
+
+/obj/item/clothing/mask/gas/borg/proc/futile()
+	if(world.time >= saved_time + cooldown2)
+		saved_time = world.time
+		var/sound = pick('DS13/sound/effects/borg/withstandassault.ogg','DS13/sound/effects/borg/dontresist.ogg')
+		var/phrase_text = "Resistance is futile."
+		src.audible_message("<font color='green' size='4'><b>[phrase_text]</b></font>")
+		playsound(src.loc,sound, 100, 0, 4)
+
+/obj/item/clothing/suit/space/borg
+	name = "borg exoskeleton"
+	desc = "A heavily armoured exoskeletal system held in place via multiple attachment points which are embedded painfully into the skin."
+	icon = 'DS13/icons/obj/suits.dmi'
+	alternate_worn_icon = 'DS13/icons/mob/suit.dmi'
+	icon_state = "borg"
+	item_state = null
+	body_parts_covered = FULL_BODY
+	cold_protection = FULL_BODY
+	flags_inv = HIDEGLOVES | HIDESHOES | HIDEJUMPSUIT
+	slowdown = 2
+	item_flags = NODROP
+	clothing_flags = STOPSPRESSUREDAMAGE | THICKMATERIAL | SHOWEROKAY
+	armor = list(melee = 40, bullet = 5, laser = 15, energy = 0, bomb = 15, bio = 100, rad = 70) //they can't react to bombs that well, and emps will rape them
+	resistance_flags = FIRE_PROOF
+	allowed = list(/obj/item/flashlight)
+	heat_protection = FULL_BODY
+	max_heat_protection_temperature = SPACE_SUIT_MAX_TEMP_PROTECT
+	cold_protection = FULL_BODY
+	min_cold_protection_temperature = SPACE_SUIT_MIN_TEMP_PROTECT
+	permeability_coefficient = 0.01
+	gas_transfer_coefficient = 0.01
+	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	var/footstep = 1
+	var/datum/component/mobhook
+
+/obj/item/clothing/suit/space/borg/IsReflect() //Watch your lasers, they adapt quickly
+	if(prob(GLOB.borg_collective.adaptation))
```

Comment:
Can the probability get to 100% blocking?

Reply:
I haven't observed this, but I believe so